fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5657,14 +5741,68 @@ class VersionedBTree {
 		// Try to add a record of the given delta size to the page.
 		// If force is true, the page will be expanded to make the record fit if needed.
 		// Return value is whether or not the record was added to the page.
-		bool addRecord(const RedwoodRecordRef& rec, int deltaSize, bool force) {
+		bool addRecord(const RedwoodRecordRef& rec,
+		               int deltaSize,
+		               bool force,
+		               unsigned int height,
```

Comment:
Since all of these new arguments are only used if `useEncryptionDomain` is true, it seems like a good style choice to put that argument first in the set.  Also, it seems that all except `next` are generated in `splitPages()` before the loop so they do not change from call to call, so maybe they could be members instead?

Reply:
It looks to me everything existing in PageToBuild are the state of the page, so I was trying to avoid putting states unrelated to the particular page into the struct. But let me change that.